### ISSUE TITLE
Local state is unreachable inside FirestoreMutation render function

### ISSUE THREAD

**Issue by MarcCoet** [rakannimer/react-firebase#14]
In a component, I create some state. I return my jsx with a `FirestoreMutation` and try to access that state inside it but it is not getting through.
I tried passing what I need as prop to `FirestoreMutation` in case you are passing unknown props to the child render function but apparently you are not.
Maybe I am missing something obvious because I didn't have time to give it a proper look  but it seems local state is unreachable inside `FirestoreMutation`
Cheers

**Comment by rovervannispen** [comment]
Same issue! A fix would be highly appreciated

**Comment by drugoi7** [comment]
Did you find a solution or a workaround?

**Comment by m4rrc0** [comment]
Nope. Went back to my own implementation of firebase with react.

**Comment by x5engine** [comment]
shit i was struggling for hours with this and now just saw this! shit

**Comment by davidgamero** [comment]
Yup I've got this same issue going on I've been struggling with it for days

**Comment by davidgamero** [comment]
In case this helps someone in the future, I did what @m4rrc0 suggested and added my own firebase implementation for my mutations.
I referenced the [FirebaseDatabaseMutation.tsx](https://github.com/rakannimer/react-firebase/blob/master/modules/database/src/components/FirebaseDatabaseMutation.tsx) source.
My changes: 
```
// Import firebase again instead of FirebaseDatabaseMutation
import { firebase } from '@firebase/app';
...

    // Inside my function component
    const pushMessage = (value) => {
        const path = 'messages/general';
        
        // Get ref to my database path (in my case for pushing) 
        const firebaseRef = firebase
          .app()
          .database()
          .ref(path);
    
        // Returns promise, need to add types for typescript to clean it up, but this worked
        return firebaseRef.push(value);
    }
```

**Comment by ivanhoe-dev** [comment]
I'm still getting this issue after 2 years!
